test/dialect/postgresql/test_reflection.py

```diff
@@ -2197,6 +2197,30 @@ def test_reflect_with_not_valid_check_constraint(self):
             ],
         )
 
+    def test_reflect_with_no_inherit_check_constraint(self):
+        rows = [
+            ("foo", "some name", "CHECK ((a IS NOT NULL)) NO INHERIT", None)
+        ]
+        conn = mock.Mock(
+            execute=lambda *arg, **kw: mock.MagicMock(
+                fetchall=lambda: rows, __iter__=lambda self: iter(rows)
+            )
+        )
+        check_constraints = testing.db.dialect.get_check_constraints(
+            conn, "foo"
+        )
+        eq_(
+            check_constraints,
+            [
+                {
+                    "name": "some name",
+                    "sqltext": "a IS NOT NULL",
+                    "dialect_options": {"no_inherit": True},
```

Comment:
I fail to see how the provided code would return these dialect options

Reply:
my previous comment is probably not clear. I meant that the code seems to be missing an if to set this value in the dialect options